### ISSUE TITLE
Any ways to make bgset plugin also do progressive load?

### ISSUE THREAD

**Issue by bitinn** [aFarkas/lazysizes#124]
Sort of like the progressive plugin, but AFAIK that plugin only works with `img` tags right? I don't see the progressive jpeg effect when combining progressive plugin with bgset plugin, eg. hi-res background show up only when fully loaded.

I guess it because bgset load the hi-res image through an `img` element, so it suffers from the same problem [described by progressive plugin](https://github.com/aFarkas/lazysizes/tree/gh-pages/plugins/progressive#how-it-works) ?

PS: I am using LQIP pattern with background image (a hero image) by setting the smallest image in CSS, and use `lazysizes` to load the hi-res version.

**Comment by bitinn** [comment]
I updated my strategy to do following:

1. use media query to specific the minimum resolution on different css width (disregard pixel density)
2. then use lazysizes bgset to load the optimal resolution.
3. now make sure all my images are progressive, then the initial load will be progressive, and bgset not showing the progressive effect is less a problem.

We can also use `img` and it has advantage of loading faster (browser delay background-image download); but we will need `object-fit` polyfill to achieve `background-size: cover` effect on many browsers and it's not semantically correct as our hero image is visual decoration.

TL;DR, my issue is probably not a big deal, just a note that bgset doesn't show progressive jpeg like default background-image.

**Comment by aFarkas** [comment]
> ...that plugin only works with img tags right?

yes.

> I guess it because bgset load the hi-res image through an img element, so it suffers from the same problem described by progressive plugin ?

No this is due to the fact, that the ``currentSrc`` property of the responsive image API only returns the selected image candidate after it was fully downloaded and not after it was selected. If I would get the selected image request/pending image request earlier, I could implement this.

> We can also use img and it has advantage of loading faster (browser delay background-image download); but we will need object-fit polyfill to achieve background-size: cover effect on many browsers and it's not semantically correct as our hero image is visual decoration.

Absolutely right. If you have just one hero image, that is always above the fold, this should be the best way from a pure performance aspect. (You can add ``role="presentation" alt=""`` to modify the semantics of you html).
